### PR TITLE
Move device parser utility function

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Addresses comment https://github.com/PyTorchLightning/pytorch-lightning/pull/9987#discussion_r738686821

The Trainer and Lite share a device parsing logic. We can move this static method out of Trainer and place it where the rest of the device parsing lives. This way, Lite does not need to import the Lightning Trainer.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)
   Changelog not needed, the function was protected/internal

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

### CODE REVIEW

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -1426,20 +1426,6 @@ def call_hook(
...
-    def _parse_devices(
-        gpus: Optional[Union[List[int], str, int]],
-        auto_select_gpus: bool,
-        tpu_cores: Optional[Union[List[int], str, int]],
-    ) -> Tuple[Optional[List[int]], Optional[Union[List[int], int]]]:
-        if auto_select_gpus and isinstance(gpus, int):
-            gpus = pick_multiple_gpus(gpus)
-
-        # TODO (@seannaren, @kaushikb11): Include IPU parsing logic here
-        gpu_ids = device_parser.parse_gpu_ids(gpus)
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply pytorch_lightning/trainer/trainer.py:1468]: resolved

File: pytorch_lightning/utilities/device_parser.py [hunk 2]
```diff
@@ -48,6 +49,20 @@ def determine_root_gpu_device(gpus: List[int]) -> Optional[int]:
     return root_gpu
 
 
+def _parse_devices(
+    gpus: Optional[Union[List[int], str, int]],
+    auto_select_gpus: bool,
+    tpu_cores: Optional[Union[List[int], str, int]],
+) -> Tuple[Optional[List[int]], Optional[Union[List[int], int]]]:
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply pytorch_lightning/utilities/device_parser.py:56]: Sounds good! How do you like  #10265 ?

**Review by carmocca — APPROVED** [review]

**Review by kaushikb11 — CHANGES_REQUESTED** [review]
@awaelchli I had discussed to keep the Trainer method, and use the utility function inside it

**Comment by awaelchli** [comment]
Changed

**Review by justusschock — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**